src/rust/pipeline-ingress/tests/integration_test.rs

```diff
@@ -118,36 +114,43 @@ async fn test_publish_raw_log_sends_message_to_kafka(
   </EventData>
 </Event>
 "#.into();
-    let expected_log_event = log_event.clone();
 
-    let kafka_scanner = KafkaTopicScanner::new(ctx.consumer_config.clone())?
-        .contains(move |envelope: Envelope<RawLog>| -> bool {
-            let envelope_tenant_id = envelope.tenant_id();
-            let envelope_event_source_id = envelope.event_source_id();
-            let raw_log = envelope.inner_message();
+    let kafka_scanner = KafkaTopicScanner::new(
+        ConsumerConfig::with_topic(CONSUMER_TOPIC),
+        Duration::from_secs(30),
+    );
 
-            tracing::debug!(message = "consumed kafka message");
+    let handle = kafka_scanner
+        .contains_for_tenant(tenant_id, 1, |_: RawLog| true)
+        .await;
 
-            envelope_tenant_id == tenant_id
-                && envelope_event_source_id == event_source_id
-                && raw_log.log_event() == expected_log_event
-        })
-        .await?;
+    tracing::info!(
+        message = "sending publish_raw_log request",
+        tenant_id =% tenant_id,
+        event_source_id =% event_source_id,
+    );
```

Comment:
doesn't the span fill in those two things for us already?

Reply:
The span only exists in places where we've consumed an Envelope<T> from Kafka. It's automatically applied to everything that runs in the StreamProcessor's closure, but that's about it. Here we don't have it yet because we don't consume anything (that happens in the topic scanner).